src/fstab-generator/fstab-generator.c

```diff
@@ -578,8 +578,39 @@ static int parse_fstab(bool initrd) {
                         if (streq(canonical_where, where))
                                 canonical_where = mfree(canonical_where);
                         else
-                                log_debug("Canonicalized what=%s where=%s to %s",
-                                                what, where, canonical_where);
+                                log_debug("Canonicalized where=%s to %s (for what=%s)",
+                                                where, canonical_where, what);
+                }
+
+                if (fstab_test_option(me->mnt_opts, "bind\0" "rbind\0")) {
+                        _cleanup_free_ char
+                                *absolute_what = NULL,
+                                *canonical_what = NULL;
+
+                        path_kill_slashes(what);
+
+                        /* Both mount and fstab-generator have accepted relative paths.
```

Comment:
i wouldn't mind if we would become stricter here, and would refuse non-absolute paths in /etc/fstab, except for the swap case (which is the only reason we support that afaict)